### PR TITLE
Fix #6665 by re-adding a few used toStrings

### LINKED ISSUE #6665: `Entity` requires a hand-implemented `toString`, because it's part of `CodeSmell.toString`
## Steps to Reproduce
Put `findings.toString()` anywhere, e.g. test results HTML.

## Expected Behavior
`.toString()` works as before.

I think it should be considered public API and be tested.

## Observed Behavior
```
Found findings:
CodeSmell(issue=Issue(id='UptightFun'), entity=io.gitlab.arturbosch.detekt.api.Entity@5b88b8e, message=Bad code!, references=[], severity=ERROR, id='UptightFun')
CodeSmell(issue=Issue(id='UptightFun'), entity=io.gitlab.arturbosch.detekt.api.Entity@58253c57, message=Bad code!, references=[], severity=ERROR, id='UptightFun')
```
There's no way to know what that entity is without actually debugging the code, or writing a good amount of code to generate external toStrings.

## Context
This change caused the problem: https://github.com/detekt/detekt/pull/6477/files#diff-33451aa64de4e366a232a4eaa18ae7f23aa92e389ca711c85eb3bfb18734d11cR14

cc @3flex @BraisGabin

## Your Environment

* Version of detekt used: `main-SNAPSHOT` (e9423cb)
* Version of Gradle used (if applicable): N/A
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): https://github.com/TWiStErRob/net.twisterrob.detekt/pull/11/checks?check_run_id=19158201624
Example: `fails on multiple findings() (net.twisterrob.detekt.testing.VerifyTest$verifyNoFindings function) failed`

### PR DESCRIPTION
**Pull request by TWiStErRob**
* Fixes #6665

See reasoning in #6665.

### CODE REVIEW

File: detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt [hunk 1]
```diff
@@ -55,6 +55,9 @@ class FilePath constructor(
         }
     }
 
+    override fun toString(): String =
```
Comment (TWiStErRob) [inline detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt:58]: Can someone please help to make the API checks pass on CI? (feel free to push or tell me what to do)

Comment (atulgpt) [reply detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt:58]: command that I use before pushing anything which generate the API dumb as well `./gradlew detektMain detektTest detektTestFixtures apiDump generateDocumentation compileKotlin compileTestKotlin compileTestFixturesKotlin -PwarningsAsErrors=true --parallel  --no-build-cache --rerun-tasks --stacktrace`

Comment (TWiStErRob) [reply detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt:58]: nice list, feels like there should be a `gradlew prePush`

File: detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt [hunk 2]
```diff
@@ -39,7 +39,7 @@ fun PsiFile.absolutePath(): Path = absolutePath ?: Path(virtualFile.path)
 /**
  * Represents both absolute path and relative path if available.
  */
-class FilePath constructor(
+class FilePath(
```
Comment (atulgpt) [inline detekt-psi-utils/src/main/kotlin/io/github/detekt/psi/KtFiles.kt:42]: good candidate for rule https://github.com/detekt/detekt/issues/6637

File: detekt-psi-utils/src/test/kotlin/io/github/detekt/psi/KtFilesSpec.kt [hunk 3]
```diff
@@ -48,4 +49,26 @@ class KtFilesSpec {
...
+        private val ps = File.separator
+
+        @Test fun `toString of absolute path`() {
+            val filePath = FilePath.fromAbsolute(File("/a/b/c").toPath())
+
+            assertThat(filePath.toString())
+                .isEqualTo("FilePath(absolutePath=${ps}a${ps}b${ps}c, basePath=null, relativePath=null)")
+        }
+
+        @Test fun `toString of relative path`() {
```
[the remark this answers is not in the document]
Comment (TWiStErRob) [reply detekt-psi-utils/src/test/kotlin/io/github/detekt/psi/KtFilesSpec.kt:66]: https://github.com/detekt/detekt/pull/6678/files/c4f2c927cba87f3bb2c1d44dcad33ce8857bdd4b#diff-334dc21acb7bb5a0e25a3f3e2789aa6bb32e903b500073684317dc52d0e73d80R19 this one has them

[the remark this answers is not in the document]
Comment (atulgpt) [reply detekt-psi-utils/src/test/kotlin/io/github/detekt/psi/KtFilesSpec.kt:66]: I am blown away 🤯. Should we also add TC for that? Or is it fine?

[the remark this answers is not in the document]
Comment (TWiStErRob) [reply detekt-psi-utils/src/test/kotlin/io/github/detekt/psi/KtFilesSpec.kt:66]: I didn't add missing tests, only for the code I added ;)
There's a lot missing in this area, they're probably covered everywhere else.

**Review by atulgpt — APPROVED** [review]

**Pull request merged**